components/match2/commons/match_group_legacy.lua

```diff
@@ -6,259 +6,364 @@
 -- Please see https://github.com/Liquipedia/Lua-Modules to contribute
 --
 
-local Legacy = {}
-
 local Arguments = require('Module:Arguments')
+local Array = require('Module:Array')
+local Class = require('Module:Class')
 local Json = require('Module:Json')
 local Logic = require('Module:Logic')
 local Lua = require('Module:Lua')
-local MatchGroup = require('Module:MatchGroup')
+local PageVariableNamespace = require('Module:PageVariableNamespace')
 local String = require('Module:StringUtils')
 local Table = require('Module:Table')
-local Variables = require('Module:Variables')
 
-local MatchGroupLegacyDefault = Lua.import('Module:MatchGroup/Legacy/Default')
+local MatchGroup = Lua.import('Module:MatchGroup')
 local MatchSubobjects = Lua.import('Module:Match/Subobjects')
-local WikiSpecific = Lua.import('Module:Brkts/WikiSpecific')
 
-local IS_USERSPACE = false
-local NAMESPACE_USER = 2
+local globalVars = PageVariableNamespace()
+
+local MAX_NUMBER_OF_OPPONENTS = 2
 local RESET_MATCH = 'RxMBR'
 local THIRD_PLACE_MATCH = 'RxMTP'
-local _args
-local _type
 
----@param frame Frame
+---@alias roundKeys {R: number, G: number, W: number, D: number}
+---@alias match1Keys {opp1: string, opp2: string, details: string}
+---@alias match2mapping {[string]: match1Keys}|{[string] : string}
+
+---@class MatchGroupLegacy
+---@operator call(table): MatchGroupLegacy
+---@field args table
+---@field bracketType string
+---@field newArgs table
+---@field mapMappings {[number]: table}
+local MatchGroupLegacy = Class.new(function(self, frame)
+	local args = Arguments.getArgs(frame)
+	assert(String.isNotEmpty(args.id), 'Argument \'id\' is empty')
+	assert(String.isNotEmpty(args.template), 'Argument \'template\' is empty')
+	assert(String.isNotEmpty(args.templateOld), 'Argument \'templateOld\' is empty')
+	assert(String.isNotEmpty(args.type), 'Argument \'type\' is empty')
+
+	self.args = args
+end)
+
+---@param match2id string
 ---@return string
-function Legacy.get(frame)
-	_args = Arguments.getArgs(frame)
-	mw.addWarning('You are editing a page that uses a Legacy Bracket. '
-		.. 'Please use the new Bracket System on new pages.')
-	local nameSpaceNumber = mw.title.getCurrentTitle().namespace
+function MatchGroupLegacy._getSimplifiedId(match2id)
```

Comment:
isn't there already a function for this in another module, maybe matchgroup util

Reply:
yap